cmake/Runner.cmake

```diff
@@ -0,0 +1,38 @@
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See http://swift.org/LICENSE.txt for license information
+# See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+
+function(SwiftTesting_GenerateDerivedSources)
+  message(STATUS "SwiftTesting: generating derived sources...")
+
+  # Calculate lowercaseprojectname_SwiftTesting_Sources.
+  string(TOLOWER ${PROJECT_NAME} PROJECT_NAME_LOWER)
+  set(${PROJECT_NAME_LOWER}_SwiftTesting_Sources
+    ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}PackageTests.derived/runner-swift-testing.swift)
+
+  # Generate the runner with SPM entry point.
+  file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}PackageTests.derived)
+  file(CONFIGURE
+    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}PackageTests.derived/runner-swift-testing.swift
+    CONTENT
+"#if canImport(Testing)
```

Comment:
Are the `canImport` guards actually needed here? The analogous code generated by SwiftPM may need to include them but would a client using this copy always be known to have this module available?

Reply:
It would depend on whether a test target built with CMake can guarantee linkage to swift-testing. If it's guaranteed, then the guard isn't needed. If it may vary from file to file (or from test target to test target within a project) then the guard may still be needed.

(Since we've asked @jeffdav to remove this file anyway, I guess it's moot.)